e2e/poms/task1.ts

```diff
@@ -0,0 +1,190 @@
+        const page2Promise = this.page.waitForEvent('popup');
+        await this.page.getByTestId(TASK1SELECTOR.testIds.publishPreviewButton).click();
+        const page2 = await page2Promise;
+
+        await expect(page2.getByTestId(TASK1SELECTOR.testIds.formMultipleChoiceOption)).toBeHidden();
+
+        const displayedOptions = await page2.getByTestId(TASK1SELECTOR.testIds.singleChoiceOption).allTextContents();
+        expect(JSON.stringify(originalOptions)).not.toBe(JSON.stringify(displayedOptions));
+
+        await this.page.getByTestId(TASK1SELECTOR.testIds.multiChoiceOptionsContainer).click();
+        await this.page.getByTestId(TASK1SELECTOR.testIds.hideQuestionToggleLabel).click();
+        await this.page.getByTestId(TASK1SELECTOR.testIds.publishButton).click();
+        const page3Promise = this.page.waitForEvent('popup');
+        await this.page.getByTestId(TASK1SELECTOR.testIds.publishPreviewButton).click();
+        const page3 = await page3Promise;
+        await page2.waitForTimeout(TASK1SELECTOR.timeouts.long);
+        expect(page3.getByText(TASK1TEXT.texts.multipleChoice)).toBeVisible();
+        await page3.waitForTimeout(TASK1SELECTOR.timeouts.medium);
+    }
+
+    async publishForm() {
+        await this.page.getByTestId(TASK1SELECTOR.testIds.publishButton).click();
+        await this.page.getByTestId(TASK1SELECTOR.testIds.moreDropdownIcon).click();
+        await this.page.getByTestId(TASK1SELECTOR.testIds.analyticsMoreTab).click();
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.visitsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText("0");
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.startsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText("0");
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.submissionsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText("0");
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.completionRateMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText("0%");
+    }
+
+    async afterVisit() {
+        const initialVisitCount = Number(
+            await this.page.getByTestId(TASK1SELECTOR.testIds.visitsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount).textContent()
+        );
+        const page1Promise = this.page.waitForEvent("popup");
+        await this.page.getByTestId(TASK1SELECTOR.testIds.publishPreviewButton).click();
+        const page1 = await page1Promise;
+        await this.page.waitForTimeout(TASK1SELECTOR.timeouts.medium);
+        await this.page.reload();
+        await expect(
+            this.page.getByTestId(TASK1SELECTOR.testIds.visitsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
+        ).toHaveText(String(initialVisitCount + 1));
+        await page1.close();
```

Comment:
`this.page.getByTestId(TASK1SELECTOR.testIds.visitsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)` is repeating in multiple places within the method. So, it makes sense to make it a constant.

```suggestion
        const insightsCount = this.page.getByTestId(TASK1SELECTOR.testIds.visitsMetric).getByTestId(TASK1SELECTOR.testIds.insightsCount)
        const initialVisitCount = Number(
            await insightsCount.textContent()
        );
        const page1Promise = this.page.waitForEvent("popup");
        await this.page.getByTestId(TASK1SELECTOR.testIds.publishPreviewButton).click();
        const page1 = await page1Promise;
        await this.page.waitForTimeout(TASK1SELECTOR.timeouts.medium);
        await this.page.reload();
        await expect(insightsCount).toHaveText(String(initialVisitCount + 1));
        await page1.close();
```

**Positive comment**
InitialVisitCount is an excellent name. Use similar naming across the codebase.